MatrixSDK/MXSession.m

```diff
@@ -926,18 +927,35 @@ - (NSString *)syncFilterId
 
 - (BOOL)isPauseable
 {
-    return _state == MXSessionStateSyncInProgress
-        || _state == MXSessionStateRunning
-        || _state == MXSessionStateBackgroundSyncInProgress
-        || _state == MXSessionStatePauseRequested;
+    switch (_state)
+    {
+        case MXSessionStateSyncInProgress:
+        case MXSessionStateRunning:
+        case MXSessionStateBackgroundSyncInProgress:
+        case MXSessionStatePauseRequested:
+        case MXSessionStateSyncError:
+        case MXSessionStateHomeserverNotReachable:
+            return YES;
+        default:
+            return NO;
+    }
 }
 
 - (BOOL)isResumable
 {
-    return !eventStreamRequest ||
-        (_state == MXSessionStateBackgroundSyncInProgress
-        || _state == MXSessionStatePauseRequested
-        || _state == MXSessionStatePaused);
+    if (!eventStreamRequest)
+    {
+        return YES;
+    }
+    switch (_state)
```

Comment:
To understand this bit, a session can only be resumed if it hasn't previously failed, right? In that case a session needs to first be reconnected?

Reply:
Well, depends on the failure. There are certain cases that session needs to be restarted (like `storeDataReady` and `initialSyncFailed`), for other failures we'll try to pause the session after them.